Applications/Spire/Include/Spire/Ui/ScrollableListBox.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef SPIRE_SCROLLABLE_LIST_BOX_HPP
+#define SPIRE_SCROLLABLE_LIST_BOX_HPP
+#include <QLayoutItem>
+#include <QWidget>
+#include "Spire/Ui/ScrollBox.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /**
+   * Represents a ScrollBox which contains a ListView.
+   */
+  class ScrollableListBox : public ScrollBox {
```

Comment:
I have been thinking about how to remove the inheritance. The big challenge is to override the update_ranges() function. I posted a substitute of the inheritance by using a signal which returns a boolean value on Asana, but I haven't gotten your feedback.

Reply:
The signal approach is too complicated. Can you come up with some alternatives?

Reply:
This issue is basically to implement a polymorphism without inheritance. What I can think of is to use a callback function.

1. Define a callback member in the ScrollBox:
```c++
// The boolean return value indicates whether the update has been performed by the callback function. 
 std::function<bool()> m_callback; 
```
And a member function:
```
Void set_callback(std::function<bool()>& callback);
```
2. Call the callback function in the entry of the ScrollBox::update_ranges:
```
void ScrollBox::update_ranges() {
  if(m_callback && m_callback()) {
     return;
  }
       …
}
```
When the callback function returns true, the ScrollBox::update_ranges function will not continue to execute; otherwise, the ScrollBox::update_ranges will be performed.

Reply:
@kranar  What do you think about it?

Reply:
Too complicated in my opinion.

Reply:
The ScrollBox::update_ranges function in the ScrollBox is called when ScrollBox gets the hide and show event of ScrollBars and the resize event of ScrollBox, so I can try to intercept these events in the ScrollableListBox and call its own update_ranges function and prevent them from being passed to the ScrollBox.